### PR TITLE
Autograder Integration [WIP]

### PR DESCRIPTION
**Pull request by Sumukh**
autograding

Autograder integration. 

Currently the integration will grade all of the final submissions - but there is nothing stopping us from doing more. There's some infrastructure work to make the solution more permanent - but this should be robust enough to play around with. 

@alvinwan - ~~Can you transition the changes from the other autograding PR into this branch?  (the ones where the scoring was moved to submissions?) (see email for more details)~~

### CODE REVIEW

**Comment by Sumukh** [comment]
There's a bit more work to do 

- on the OK side with regards to accepting grades. (I'm not happy with the current solution of passing both the submission and the backup id to the autograder - it's a bunch of extra queries and info) 
- infrastructure wise making the autograder autoscale & respawn/retry failed jobs.  
- using refresh-tokens & generating one that's class wide (from a course gmail or similiar). Using a staff members is ok - until they auth with the client 25 more times - then our token expires. A course email wouldn't be used as much.

File: server/app/api.py [hunk 1]
```diff
@@ -761,7 +779,38 @@ def invite(self, obj, user, data):
...
+        for fsub in fsubs:
+          subm_ids[fsub.submission.id()] = fsub.submission.get().backup.id()
+
+        ag_url = "http://104.154.46.183:5000"
+        data = {'subm_ids': subm_ids,
+        'assign_name': obj.display_name,
+        'starter_zip_url': obj.zip_file_url,
+        'access_token': obj.access_token,
+        'grade_script': obj.grading_script_file,
+        'testing': True}
```
Comment (Sumukh) [inline server/app/api.py]: Eventually remove this (or make it a parameter)

**Comment by Sumukh** [comment]
Just deployed this to production. Autograding works. It's submitting scores too. 

image

**Comment by albert12132** [comment]
Awesome. I'd like to try it out. Is there anything I need to be aware of
when using this feature?

**Comment by Sumukh** [comment]
Merging to have master reflect the currently deployed version.

**Pull request merged**